### ISSUE TITLE
AttributeError: Distribution instance has no attribute 'use_hg_version'

### ISSUE THREAD

**Issue by jaraco** [jaraco/jaraco.vcs#1]
Originally reported by: **Jason R. Coombs (BitBucket: [jaraco](http://bitbucket.org/jaraco), GitHub: [jaraco](http://github.com/jaraco))**

---

jps reports an error installing limpy.

```
#!traceback

Processing dependencies for limpy
Searching for jaraco.compat>=0.1

Link to http://pypi.python.org/simple/jaraco.compat/ ***BLOCKED*** by --allow-hosts

Couldn't retrieve index page for 'jaraco.compat'
Best match: jaraco.compat 0.1
Downloading http://cheeseshop.polimetrix.com/jaraco.compat-0.1.tar.gz
Processing jaraco.compat-0.1.tar.gz
Running jaraco.compat-0.1/setup.py -q bdist_egg --dist-dir /tmp/easy_install-SnZcw1/jaraco.compat-0.1/egg-dist-tmp-x697OH
Searching for hgtools>=0.4
Reading http://pypi.python.org/simple/hgtools/
Reading http://bitbucket.org/jaraco/hgtools/
Reading http://bitbucket.org/jaraco/hgtools/downloads/
Best match: hgtools 0.5
Downloading http://pypi.python.org/packages/source/h/hgtools/hgtools-0.5.zip#md5=de3002a80877e7433e32c4eb1d6d259e
Processing hgtools-0.5.zip
Running hgtools-0.5/setup.py -q bdist_egg --dist-dir /tmp/easy_install-SnZcw1/jaraco.compat-0.1/temp/easy_install-5kI63A/hgtools-0.5/egg-dist-tmp-i9QTSA
Error getting managers in hgtools.file_finder_plugin: Error when calling the metaclass bases
    first argument must be callable
zip_safe flag not set; analyzing archive contents...

Installed /tmp/easy_install-SnZcw1/jaraco.compat-0.1/hgtools-0.5-py2.6.egg
Error getting managers in hgtools.file_finder_plugin: Error when calling the metaclass bases
    cannot create 'builtin_function_or_method' instances
zip_safe flag not set; analyzing archive contents...
Adding jaraco.compat 0.1 to easy-install.pth file

Installed /home/jeff.stearns/env/marin/lib/python2.6/site-packages/jaraco.compat-0.1-py2.6.egg
Searching for zope.interface

Link to http://pypi.python.org/simple/zope.interface/ ***BLOCKED*** by --allow-hosts

Couldn't retrieve index page for 'zope.interface'
Best match: zope.interface 3.3.0
Downloading http://cheeseshop.polimetrix.com/zope.interface-3.3.0.tar.gz
Processing zope.interface-3.3.0.tar.gz
Running zope.interface-3.3.0/setup.py -q bdist_egg --dist-dir /tmp/easy_install-TSPoJy/zope.interface-3.3.0/egg-dist-tmp-1USz_5
Traceback (most recent call last):
  File "/home/jeff.stearns/env/marin/bin/easy_install", line 8, in <module>
    load_entry_point('setuptools==0.6c11', 'console_scripts', 'easy_install')()
  File "/home/jeff.stearns/env/marin/lib/python2.6/site-packages/setuptools-0.6c11-py2.6.egg/setuptools/command/easy_install.py", line 1712, in main
    with_ei_usage(lambda:
  File "/home/jeff.stearns/env/marin/lib/python2.6/site-packages/setuptools-0.6c11-py2.6.egg/setuptools/command/easy_install.py", line 1700, in with_ei_usage
    return f()
  File "/home/jeff.stearns/env/marin/lib/python2.6/site-packages/setuptools-0.6c11-py2.6.egg/setuptools/command/easy_install.py", line 1716, in <lambda>
    distclass=DistributionWithoutHelpCommands, **kw
  File "/usr/lib/python2.6/distutils/core.py", line 152, in setup
    dist.run_commands()
  File "/usr/lib/python2.6/distutils/dist.py", line 975, in run_commands
    self.run_command(cmd)
  File "/usr/lib/python2.6/distutils/dist.py", line 995, in run_command
    cmd_obj.run()
  File "/home/jeff.stearns/env/marin/lib/python2.6/site-packages/setuptools-0.6c11-py2.6.egg/setuptools/command/easy_install.py", line 211, in run
    self.easy_install(spec, not self.no_deps)
  File "/home/jeff.stearns/env/marin/lib/python2.6/site-packages/setuptools-0.6c11-py2.6.egg/setuptools/command/easy_install.py", line 446, in easy_install
    return self.install_item(spec, dist.location, tmpdir, deps)
  File "/home/jeff.stearns/env/marin/lib/python2.6/site-packages/setuptools-0.6c11-py2.6.egg/setuptools/command/easy_install.py", line 478, in install_item
    self.process_distribution(spec, dist, deps)
  File "/home/jeff.stearns/env/marin/lib/python2.6/site-packages/setuptools-0.6c11-py2.6.egg/setuptools/command/easy_install.py", line 519, in process_distribution
    [requirement], self.local_index, self.easy_install
  File "/home/jeff.stearns/env/marin/lib/python2.6/site-packages/setuptools-0.6c11-py2.6.egg/pkg_resources.py", line 563, in resolve
    dist = best[req.key] = env.best_match(req, self, installer)
  File "/home/jeff.stearns/env/marin/lib/python2.6/site-packages/setuptools-0.6c11-py2.6.egg/pkg_resources.py", line 799, in best_match
    return self.obtain(req, installer) # try and download/install
  File "/home/jeff.stearns/env/marin/lib/python2.6/site-packages/setuptools-0.6c11-py2.6.egg/pkg_resources.py", line 811, in obtain
    return installer(requirement)
  File "/home/jeff.stearns/env/marin/lib/python2.6/site-packages/setuptools-0.6c11-py2.6.egg/setuptools/command/easy_install.py", line 446, in easy_install
    return self.install_item(spec, dist.location, tmpdir, deps)
  File "/home/jeff.stearns/env/marin/lib/python2.6/site-packages/setuptools-0.6c11-py2.6.egg/setuptools/command/easy_install.py", line 476, in install_item
    dists = self.install_eggs(spec, download, tmpdir)
  File "/home/jeff.stearns/env/marin/lib/python2.6/site-packages/setuptools-0.6c11-py2.6.egg/setuptools/command/easy_install.py", line 655, in install_eggs
    return self.build_and_install(setup_script, setup_base)
  File "/home/jeff.stearns/env/marin/lib/python2.6/site-packages/setuptools-0.6c11-py2.6.egg/setuptools/command/easy_install.py", line 930, in build_and_install
    self.run_setup(setup_script, setup_base, args)
  File "/home/jeff.stearns/env/marin/lib/python2.6/site-packages/setuptools-0.6c11-py2.6.egg/setuptools/command/easy_install.py", line 919, in run_setup
    run_setup(setup_script, args)
  File "/home/jeff.stearns/env/marin/lib/python2.6/site-packages/setuptools-0.6c11-py2.6.egg/setuptools/sandbox.py", line 62, in run_setup
    lambda: execfile(
  File "/home/jeff.stearns/env/marin/lib/python2.6/site-packages/setuptools-0.6c11-py2.6.egg/setuptools/sandbox.py", line 105, in run
    return func()
  File "/home/jeff.stearns/env/marin/lib/python2.6/site-packages/setuptools-0.6c11-py2.6.egg/setuptools/sandbox.py", line 64, in <lambda>
    {'__file__':setup_script, '__name__':'__main__'}
  File "setup.py", line 86, in <module>
  File "/usr/lib/python2.6/distutils/core.py", line 152, in setup
    dist.run_commands()
  File "/usr/lib/python2.6/distutils/dist.py", line 975, in run_commands
    self.run_command(cmd)
  File "/usr/lib/python2.6/distutils/dist.py", line 994, in run_command
    cmd_obj.ensure_finalized()
  File "/usr/lib/python2.6/distutils/cmd.py", line 117, in ensure_finalized
    self.finalize_options()
  File "/home/jeff.stearns/env/marin/lib/python2.6/site-packages/setuptools-0.6c11-py2.6.egg/setuptools/command/bdist_egg.py", line 94, in finalize_options
    ei_cmd = self.ei_cmd = self.get_finalized_command("egg_info")
  File "/usr/lib/python2.6/distutils/cmd.py", line 319, in get_finalized_command
    cmd_obj.ensure_finalized()
  File "/usr/lib/python2.6/distutils/cmd.py", line 117, in ensure_finalized
    self.finalize_options()
  File "/home/jeff.stearns/env/marin/lib/python2.6/site-packages/setuptools-0.6c11-py2.6.egg/setuptools/command/egg_info.py", line 86, in finalize_options
    self.egg_version = self.tagged_version()
  File "build/bdist.linux-x86_64/egg/hgtools/__init__.py", line 384, in tagged_version
AttributeError: Distribution instance has no attribute 'use_hg_version'
```

---
- Bitbucket: https://bitbucket.org/jaraco/hgtools/issue/1

**Comment by jaraco** [comment]
_Original comment by_: **Jason R. Coombs (BitBucket: [jaraco](http://bitbucket.org/jaraco), GitHub: [jaraco](http://github.com/jaraco))**

---

This was an oversight in the way the plugin worked. Apparently if hgtools was activated for a previous package (in this case jaraco.compat), then subsequent packages might fail when they didn't specify the use_hg_tools option. The plugin should not have been activated for these subsequent packages, and as of <<changeset 7a972b87a3e0>>, they will not be.

**Comment by jaraco** [comment]
_Original comment by_: **Jason R. Coombs (BitBucket: [jaraco](http://bitbucket.org/jaraco), GitHub: [jaraco](http://github.com/jaraco))**

---

Although I tested it in development mode, this problem only appears to exhibit itself when hgtools comes as an sdist egg from the cheeseshop.

**Comment by jaraco** [comment]
_Original comment by_: **Jason R. Coombs (BitBucket: [jaraco](http://bitbucket.org/jaraco), GitHub: [jaraco](http://github.com/jaraco))**

---

Another attempt to fix #1

**Comment by jaraco** [comment]
_Original comment by_: **Jason R. Coombs (BitBucket: [jaraco](http://bitbucket.org/jaraco), GitHub: [jaraco](http://github.com/jaraco))**

---

Removing version: 0.5 (automated comment)
